wsmaster/che-core-api-machine/src/main/java/org/eclipse/che/api/machine/server/MachineManager.java

```diff
@@ -1010,21 +1010,37 @@ private void cleanup() {
 
         executor.shutdown();
 
+        boolean interrupted = false;
+        final ExecutorService destroyMachinesExecutor = Executors.newCachedThreadPool(
```

Comment:
According to [this](https://github.com/eclipse/che/pull/1184):
> Each machine must be destroyed in the separate thread

I see no difference here between `FixedThreadPool` and `CachedThreadPool`.
It even doesn't depends on number of machines: 100, 1000 ... all the same. (from `*ThreadPool` point of view)